File: Leetcode/1161 Maximum Level Sum of a Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
  public:
    int maxLevelSum(TreeNode *root) {
        queue<TreeNode *> q;
        if (root != nullptr) {
            q.emplace(root);
        }
        int level = 1, max_level_sum, max_sum_level = 0;
        while (!q.empty()) {
            int n_nodes_in_level = q.size();
            int level_sum = 0;
            for (int i = 0; i < n_nodes_in_level; ++i) {
                auto node = q.front();
                q.pop();
                level_sum += node->val;
                if (node->left != nullptr) {
                    q.emplace(node->left);
                }
                if (node->right != nullptr) {
                    q.emplace(node->right);
                }
            }
            if (max_sum_level == 0 || level_sum > max_level_sum) {
                max_level_sum = level_sum;
                max_sum_level = level;
            }
            ++level;
        }
        return max_sum_level;
    }
};

File: Leetcode/1161 Maximum Level Sum of a Binary Tree.java
class Solution {
    public int maxLevelSum(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        if (root != null) {
            queue.add(root);
        }
        int level = 1;
        int maxLevelSum = 0, maxSumLevel = 0;
        while (!queue.isEmpty()) {
            int nNodesInLevel = queue.size();
            int levelSum = 0;
            for (int i = 0; i < nNodesInLevel; ++i) {
                var node = queue.poll();
                levelSum += node.val;
                if (node.left != null) {
                    queue.add(node.left);
                }
                if (node.right != null) {
                    queue.add(node.right);
                }
            }
            if (maxSumLevel == 0 || levelSum > maxLevelSum) {
                maxLevelSum = levelSum;
                maxSumLevel = level;
            }
            ++level;
        }
        return maxSumLevel;
    }
}

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */